controllers/warehouse-controller.js

```diff
@@ -9,6 +9,26 @@ const index = async (_req, res) => {
   }
 };
 
+const findWarehouse = async (req, res) => {
+  try {
+    const warehousesFound = await knex("warehouses").where({
+      id: req.params.id,
+    });
+    if (warehousesFound.length === 0) {
+      return res.status(404).json({
+        message: `warehouse with ID ${req.params.id} not found`,
+      });
+    }
+    const warehouseData = warehousesFound[0];
+    res.json(warehouseData);
```

Comment:
shall we add good status return code for consistency?

Reply:
yes, please. We can add it after line 23. I will close this pull request and create a new request.